charts/service-deployment/templates/deployment.yaml

```diff
@@ -51,14 +51,17 @@ spec:
       - name: {{ .Values.dockerconfigjson.name }}
       {{- end }}
 
-      {{- if .Values.configMaps }}
```

Comment:
Do we not need the conditional on configMaps existing ? Here and later on with the VolumeMounts ?

Reply:
Nope we don't I validated that even with both empty it deploys just fine.  Looks like an overzealous protection initially was put in.